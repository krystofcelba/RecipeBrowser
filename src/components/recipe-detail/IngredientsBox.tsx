import React from 'react';
import { StyleSheet, View, Text, StyleProp, ViewStyle } from 'react-native';

import { StylesConstants, Colors } from 'src/assets/constants';
import i18n from 'src/assets/i18n';
import { Ingredient } from 'src/services/api';

interface Props {
  style: StyleProp<ViewStyle>;
  ingredients: [Ingredient];
  seasonings: [string];
}

const IngredientsBox = ({ style, ingredients, seasonings }: Props) => (
  <View style={[styles.container, style]}>
    <Text style={styles.heading}>{i18n.t('ingredients')}</Text>
    {ingredients.map(({ amount, unit, name }, i) => (
      <View style={styles.ingredientTextContainer} key={i}>
        <Text style={styles.ingredientText}>{'\u2022'}</Text>
        <Text style={styles.ingredientText}>{`${amount} ${unit} ${name}`}</Text>
      </View>
    ))}
    <View style={styles.seasoningsContainer}>
      <Text style={styles.seasoningsTitle}>Seasonings</Text>
      <Text style={styles.seasoningsText}>
        {seasonings.map((seasoning, i) => `${seasoning}${i < seasonings.length - 1 ? ', ' : ''}`)}
      </Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  },
  heading: {
    fontSize: StylesConstants.detailBoxHeadingFontSize,
    fontWeight: 'bold',
    fontFamily: StylesConstants.fontFamily,
  },
  ingredientTextContainer: { flexDirection: 'row' },
  ingredientText: { fontFamily: StylesConstants.fontFamily, paddingLeft: StylesConstants.defaultMargin / 2 },
  seasoningsContainer: {
    flexDirection: 'column',
    marginTop: StylesConstants.defaultMargin / 2,
    marginRight: StylesConstants.defaultMargin,
  },
  seasoningsTitle: { fontFamily: StylesConstants.fontFamily, fontWeight: 'bold' },
  seasoningsText: { fontFamily: StylesConstants.fontFamily, paddingLeft: StylesConstants.defaultMargin / 2 },
});

export default IngredientsBox;
