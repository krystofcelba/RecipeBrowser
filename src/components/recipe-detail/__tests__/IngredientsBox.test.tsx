import React from 'react';
import { shallow } from 'enzyme';
import IngredientsBox from '../IngredientsBox';

const mockedRecipes = require('../../../services/__mockData__/recipes.json');

describe('IngredientsBox component', () => {
  const mockedRecipe = mockedRecipes[2];
  it('renders as expected', () => {
    const wrapper = shallow(
      <IngredientsBox style={{}} ingredients={mockedRecipe.ingredients} seasonings={mockedRecipe.seasonings} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
