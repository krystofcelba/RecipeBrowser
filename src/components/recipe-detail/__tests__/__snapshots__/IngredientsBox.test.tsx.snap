// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsBox component renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "Avenir",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Ingredients
  </Text>
  <View
    key="0"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      200 g white fish fillet (cod, plaice, sole; fresh or frozen)
    </Text>
  </View>
  <View
    key="1"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      0.5  avocados
    </Text>
  </View>
  <View
    key="2"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      0.5  grapefruits
    </Text>
  </View>
  <View
    key="3"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      1  chopped jalapeño
    </Text>
  </View>
  <View
    key="4"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      0.5 bunches coriander
    </Text>
  </View>
  <View
    key="5"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      0.5  limes (juice)
    </Text>
  </View>
  <View
    key="6"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      100 g chopped cabbage
    </Text>
  </View>
  <View
    key="7"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      •
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      1 tsp coconut oil
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "marginRight": 10,
        "marginTop": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "fontWeight": "bold",
        }
      }
    >
      Seasonings
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontFamily": "Avenir",
          "paddingLeft": 5,
        }
      }
    >
      Chili flakes, 
      unrefined salt, 
      black pepper
    </Text>
  </View>
</View>
`;
